traitsui/tests/test_actions.py

```diff
@@ -85,7 +86,13 @@ def _qt_click_button(ui):
     button.click()
 
 
-class TestActions(unittest.TestCase):
+class TestActions(BaseTestMixin, unittest.TestCase):
+
+    def setUp(self):
+        BaseTestMixin.setUp(self)
+
+    def tearDown(self):
+        BaseTestMixin.tearDown(self)
```

Comment:
Do we need to override these? This is more of a general python question, if a subclass inherits from 2 base classes that have a method of the same name, which method is used when it is called on a subclass?  In this case I believe the `unittest.TestCase` setUp and tearDown methods do nothing by default.  Im guessing by the fact that you included this, that in this case the programmer would need to specify by overwriting the method?

Reply:
For `TestActions`, I can totally do this instead:
```
class TestActions(BaseTestMixin, unittest.TestCase):
    # setUp and tearDown are not overridden.
    # BaseTestMixin.setUp and BaseTestMixin.tearDown are called by virtue of inheritance.
```
This would be similar to the tests that previously only subclass `GuiTestAssistant` and `TestCase`. This works the same way:
```
class TestSomething(GuiTestAssistant, unittest.TestCase):
    # GuiTestAssistant.setUp and GuiTestAssistant.tearDown by virtue of inheritance.
```
But as soon as I add another base class that also offers `setUp` and `tearDown`, I need to expand the subclass's `setUp` and `tearDown` to call all base classes' setUp/tearDown.

e.g. as soon as I add `BaseTestMixin`, I need to remember to call `GuiTestAssistant.setUp` and `GuiTestAssistant.tearDown`:
```
class TestSomething(BaseTestMixin, GuiTestAssistant, unittest.TestCase):
    def setUp(self):
        BaseTestMixin.setUp(self)
        GuiTestAssistant.setUp(self)

    def tearDown(self):
        GuiTestAssistant.tearDown(self)
        BaseTestMixin.tearDown(self)
```

If I did not do this, i.e. if I had done this instead:
```
class TestSomething(BaseTestMixin, GuiTestAssistant, unittest.TestCase):
    # setUp and tearDown are not overridden
```
Only `BaseTestMixin.setUp` and `BaseTestMixin.tearDown` will be called. `GuiTestAssistant.setUp` and `GuiTestAssistant.tearDown` that were previously called will NOT be called. That would be an error.

This is why even though I do not need to override `TestActions.setUp` and `TestActions.tearDown`, I still override them to call `BaseTestMixin`'s setUp and tearDown so that the next person adding another base class doesn't need to think that hard. The computer sees the same thing, but this taxes the human less and lowers the risk of human errors.

Reply:
>This is why even though I do not need to override TestActions.setUp and TestActions.tearDown, I still override them to call BaseTestMixin's setUp and tearDown so that the next person adding another base class doesn't need to think that hard. The computer sees the same thing, but this taxes the human less and lowers the risk of human errors.

Personally I wouldn't have done this because it's not common to add baseclasses (or change baseclasses) to testcases.

Reply:
Yeah I tend to think this is closer to a personal preference which I won't ask others to do, but I'd do myself :)